### ISSUE TITLE
Add `ignoreSelectors: []` to `selector-max-compound-selectors`

### ISSUE THREAD

**Issue by FloEdelmann** [stylelint/stylelint#7536]
### What is the problem you're trying to solve?

Vue 2[^vue3] uses a special syntax to allow component-scoped styles to affect child components:

[^vue3]: In Vue 3, the syntax was changed to `.foo :deep(.bar)`, see <https://vuejs.org/api/sfc-css-features.html#deep-selectors>, and the last "pass" example.

```css
/* all equivalent: */
.foo >>> .bar {}
.foo::v-deep .bar {}
.foo ::v-deep .bar {}
.foo /deep/ .bar {}
```

See https://vue-loader.vuejs.org/guide/scoped-css.html#deep-selectors.

When using Sass, `>>>` and `/deep/` are not supported correctly, so we are stuck with using `::v-deep`. However, that "fake" selector counts towards the [`selector-max-compound-selectors`](https://stylelint.io/user-guide/rules/selector-max-compound-selectors/) limit.

### What solution would you like to see?

Since that "fake" selector rather acts as a combinator, I'd like to ignore it in the `selector-max-compound-selectors` rule, while keeping the current limit:

#### Rule

```json
"selector-max-compound-selectors": [3, { "ignore": ["::v-deep"] }]
```

#### Fail

```css
.foo .bar .baz ::v-deep .lorem {}

.foo .baz {
  & ::v-deep > .bar .lorem {}
}
```

#### Pass

```css
#foo ::v-deep #bar > #baz {}

.foo .baz {
  & ::v-deep > .bar {}
}

.foo + div :deep(a b ~ c) {} /* `a b ~ c` is inside `:deep()`, so it is evaluated separately */
```

**Comment by ybiquitous** [comment]
@FloEdelmann Thanks for the proposal. Adding a new `ignoreSelectors` option like [`selector-no-vendor-prefix`](https://stylelint.io/user-guide/rules/selector-no-vendor-prefix/#ignoreselectors-regex-regex-non-regex) makes sense to me. 👍🏼

**Comment by ybiquitous** [comment]
@FloEdelmann Thanks for the pull request. But let's wait for other thoughts in a few days. We might have missed something.

**Comment by Mouvedia** [comment]
Angular had something similar: `::ng-deep`.  
Also if you are using `::deep` to style a part of an element's shadow tree before rendering is finished, this new option would permit to skip it.  
It also might be useful when you have generated code that produce too much specificity.  
e.g. `tr` tag
Another example that comes to mind are sass mixins that produce a lot of repetition.
e.g. `["/\.container-.*/"]` for bootstrap

tl;dr: I have found enough use cases to warrant the addition.

**Comment by Mouvedia** [comment]
I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.
